### PR TITLE
Remove reliance on external server to test providers

### PR DESCRIPTION
**Pull request by axelboc**
I've set up two Python projects inside a new `support` folder at the root of the monorepo: one to create `sample.h5` and one to start an h5grove tornado server. I've then changed the API tests as follows:

- `h5wasm-api.test.ts` now looks for the sample file at `support/sample/dist/sample.h5`
- `h5grove-api.test.ts` now connects to the h5grove support instance at http://localhost:8888, which serves the sample file from `support/sample/dist`.

Both support projects are managed with Poetry. I've added pnpm scripts for convenience and documented everything in `CONTRIBUTING.md`. Concerning the Python version, I've set it to `3.12` with a `.python-version` file at the root of the monorepo, and I explain how to use `pyenv` in the contributing guide.

Of course, I've also updated the CI `test` job with steps to set up Python, [install Poetry](https://github.com/snok/install-poetry), create/cache the Poetry environments, create the sample file and start h5grove. It's surprisingly fast; once everything is cached (including the installation of Poetry itself), it adds only a few seconds to the `test` job, which is awesome.

### CODE REVIEW

File: packages/app/src/providers/h5grove/h5grove-api.test.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ import { expect, test } from 'vitest';
 import { getValueOrError } from '../utils';
 import { H5GroveApi } from './h5grove-api';
 
-const H5GROVE_URL = 'https://bosquet.silx.org/h5grove';
+const H5GROVE_URL = 'http://localhost:8888'; // when running `pnpm support:h5grove`
 const TEST_FILE = 'sample.h5';
```
Comment (axelboc) [inline packages/app/src/providers/h5grove/h5grove-api.test.ts:13]: I'm still planning on moving this to a `.env` file once [this Vitest issue](https://github.com/vitest-dev/vitest/issues/2117) is resolved

File: .github/workflows/approve-snapshots.yml [hunk 2]
```diff
@@ -52,7 +52,7 @@ jobs:
         run: pnpm build
 
       - name: Serve ⚡️
-        run: pnpm serve & pnpm dlx wait-on http://localhost:5173
+        run: pnpm serve & pnpm dlx wait-on http://localhost:5173 -t 1m
```
Comment (axelboc) [inline .github/workflows/approve-snapshots.yml:55]: Making sure `wait-on` never waits indefinitely....

**Review by loichuder — APPROVED** [review]
I am not sure if it is worth the effort but go ahead :wink:

File: support/h5grove/pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+[tool.poetry]
+name = "h5grove-tornado"
+version = "0.1.0"
+package-mode = false
+description = "h5grove tornado server for testing purposes"
+authors = ["NA"]
+
+[tool.poetry.dependencies]
+python = "^3.10"
+h5grove = "2.0.1b0"
```
[the remark this answers is not in the document]
Comment (axelboc) [reply support/h5grove/pyproject.toml]: I had to run `poetry add h5grove[tornado]` as it requires a special syntax, but it works, good call.

[the remark this answers is not in the document]
Comment (axelboc) [reply support/h5grove/pyproject.toml]: Arf, nope, I get `ModuleNotFoundError: No module named 'typing_extensions'` when trying to start the h5grove server.

[the remark this answers is not in the document]
Comment (loichuder) [reply support/h5grove/pyproject.toml]: Weird. `typing-extensions` is in the requirements :thinking:

File: .github/workflows/lint-test.yml [hunk 4]
```diff
@@ -75,6 +75,38 @@ jobs:
...
+
+      - name: Create Poetry environments 📜
+        if: steps.poetry-venvs.outputs.cache-hit != 'true'
+        run: pnpm support:setup
+
+      - name: Create sample HDF5 file 🎨
+        run: pnpm support:sample
+
+      - name: Start h5grove support server 🔌
+        run: pnpm support:h5grove & pnpm wait-on "http-get://localhost:8888/attr/?file=sample.h5&path=/" -t 1m
```
[the remark this answers is not in the document]
Comment (axelboc) [reply .github/workflows/lint-test.yml]: Done!

File: support/h5grove/pyproject.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (axelboc) [reply support/h5grove/pyproject.toml]: I've added the dependency explicitly :shrug:

**Pull request merged**